Makefile.prow

```diff
@@ -1,92 +1,90 @@
+# currently installed binaries in GOBIN, files in GOPATH/src
+#
+# To update formatting, run:
 # GOFLAGS=-mod=mod make update
 
-GOFLAGS=-mod=mod
-CONGEN_VERSION=0.14.0
-CODEGEN_VERSION=0.22.2
-PROWBIN=/tmp/prowbin
-# Name of this Makefile
-MAKEFILE=Makefile.prow
-# emulate as close as possible upstream ci target
+# Configuration Variables
+GOFLAGS := -mod=mod
+CONGEN_VERSION := 0.14.0
+CODEGEN_VERSION := 0.22.2
+PROWBIN := /tmp/prowbin
+
+# GOPATH Setup
+GOPATH := $(shell go env GOPATH)
+GOBIN := $(GOPATH)/bin
+GOSRC := $(GOPATH)/src
+
+
 # upstream ci target: verify-modules verify all test
-#  we only need to modify verify, test, all to avoid docker usage
+# we need to modify verify, test, all to avoid usage of docker CLI
 .PHONY: ci
-ci:
-	@echo "go version is: $(shell go version)"
-	GOFLAGS=$(GOFLAGS) make verify-modules
-	make -f $(MAKEFILE) verify all test
+ci: verify-modules verify all test
+
+.PHONY: verify-modules
+verify-modules:
+	@echo "verify-modules target: calls upstream Makefile verify-modules"
+	PATH=$(PROWBIN):$(PATH) GOFLAGS=$(GOFLAGS) make verify-modules
 
 .PHONY: verify
-verify: goimports controller-gen kubectl #code-generator
-# add PROWBIN to PATH
-	@echo Verifying with PATH=$(PROWBIN):$(PATH)
-# Per
-# https://github.com/vmware-tanzu/velero/blob/dd660882d0db96d430547f39dc3694d1c1bc19f3/Makefile#L160-L163
-# code-generator tools require project to be in heirarchy such as github.com/vmware-tanzu/velero
-#  so we need to copy the project to GOPATH/src/github.com/vmware-tanzu/velero
-#  and then run verify from there
-#  otherwise the code-generator tools will fail
-	mkdir -p $(GOSRC)/github.com/vmware-tanzu/
-	cp -r . $(GOSRC)/github.com/vmware-tanzu/velero
-	cd $(GOSRC)/github.com/vmware-tanzu/velero && \
+verify: setup-env goimports controller-gen
+	@echo "Running verification scripts"
 	PATH=$(PROWBIN):$(PATH) GOFLAGS=$(GOFLAGS) hack/verify-all.sh
 
+# Build targets
 .PHONY: all
 all:
-	GOFLAGS=$(GOFLAGS) make local
-	GOFLAGS=$(GOFLAGS) BIN=velero-restore-helper make local
+	@echo "Running all targets"
+	PATH=$(PROWBIN):$(PATH) GOFLAGS=$(GOFLAGS) make local
+	PATH=$(PROWBIN):$(PATH) GOFLAGS=$(GOFLAGS) BIN=velero-restore-helper make local
 
 .PHONY: test
-# our test is modified to avoid docker usage
-test: envtest
-	@echo Testing with KUBEBUILDER_ASSETS=$(KUBEBUILDER_ASSETS)
+test: setup-env
+	# Kubebuilder Setup
+	@echo "Setting up Kubebuilder assets..."
+	$(eval KUBEBUILDER_ASSETS := $(shell $(GOBIN)/setup-envtest use -p path | sed 's/ /\\ /g'))
+	@echo "KUBEBUILDER_ASSETS is set to $(KUBEBUILDER_ASSETS)"
+	@echo "Running tests..."
 	KUBEBUILDER_ASSETS=$(KUBEBUILDER_ASSETS) GOFLAGS=$(GOFLAGS) hack/test.sh
 
-GOPATH:=$(shell go env GOPATH)
-GOBIN:=$(GOPATH)/bin
-GOSRC:=$(GOPATH)/src
-#  if KUBEBUILDER_ASSETS contains space, escape it
-KUBEBUILDER_ASSETS=$(shell echo $(shell $(GOBIN)/setup-envtest use -p path) | sed 's/ /\\ /g')
-.PHONY: envtest
-envtest: $(GOBIN)/setup-envtest
+.PHONY: lint
```

Comment:
I would remove this

we are adding without using it

Reply:
I want to introduce prow job for that target.

Reply:
on OADP, we have everything under unit-test job

would do the same approach here

Reply:
This is very much different from the upstream and usually lint is not part of unit. Having it separate gives clean information that the code is OK, just needs reformatting without checking job logs.